mne_bids/read.py

```diff
@@ -700,7 +722,36 @@ def get_matched_empty_room(bids_basename, bids_root):
             space=params.get('space', None)
         )
 
-    return er_basename
+        if 'ses' in params:  # Try to extract date from filename.
+            try:
+                meas_date = datetime.strptime(params['ses'], '%Y%m%d')
+            except (ValueError, TypeError):
+                # There is a session in the filename, but it doesn't encode a
+                # valid date.
+                pass
+
+        if meas_date is None:  # No luck so far! Let's check info['meas_date']
+            _, ext = _parse_ext(path.name)
+            if ext == '.fif':
+                extra_params = dict(allow_maxshield=True)
+            else:
+                extra_params = None
+
+            raw = read_raw_bids(bids_basename=er_basename,
+                                bids_root=bids_root,
+                                extra_params=extra_params)
+
+            meas_date = raw.info['meas_date']
+            if meas_date is None:  # There's nothing we can do.
+                continue
+
+        meas_date = meas_date.replace(tzinfo=ref_date.tzinfo)
+        delta_t = meas_date - ref_date
+        if abs(delta_t.total_seconds()) < min_delta_t:  # XXX What about ties?
```

Comment:
If it's a tie, I think selecting either is fine

Reply:
You're probably right. But maybe we should emit a warning?

Reply:
Emitting a warning now.